packages/front-end/components/PowerCalculation/stats.ts

```diff
@@ -223,85 +251,81 @@ export function findMde(
-  const c_star = mean ** 2 - z ** 2 * sigma2;
+  const a_star = 1 - (z ** 2 * sigma2) / m ** 2;
+  const b_star = -2 * m;
+  const c_star = m ** 2 - z ** 2 * sigma2;
   const disc = b_star ** 2 - 4 * a_star * c_star;
   const sol_1 = (-b_star + Math.sqrt(disc)) / (2 * a_star);
   //const sol_2 = (-b_star - Math.sqrt(disc)) / (2 * a_star);
   const mdeResults: MDEResults = {
     type: "success",
-    mde: (sol_1 - mean) / mean,
+    mde: (sol_1 - m) / m,
   };
   return mdeResults;
 }
 
 export function powerMetricWeeks(
-  powerSettings: PowerCalculationParams
+  powerSettings: PowerCalculationParams | PowerCalculationParamsBayesian
 ): PowerCalculationResults {
-  const metrics = powerSettings.metrics;
   const sampleSizeAndRuntimeNumeric: number[] = []; //for each metric, the first week they achieve 80% power.
-  const nWeeks = powerSettings.nWeeks;
-  let sequentialTuningParameter = 0.0;
-  if (powerSettings.statsEngine.sequentialTesting !== false) {
-    sequentialTuningParameter = powerSettings.statsEngine.sequentialTesting;
-  }
-  function getNumberOfMetrics(params: PowerCalculationParams): number {
-    return Object.keys(params.metrics).length;
-  }
-  const nMetrics = getNumberOfMetrics(powerSettings);
-  const metricKeys = Object.keys(metrics);
   const mySampleSizeAndRuntime: {
     [id: string]: SampleSizeAndRuntime | undefined;
   } = {};
 
   const metricThresholds = {};
-  const weeks: Week[] = [...Array(nWeeks).keys()].map((idx) => ({
+  const weeks: Week[] = [...Array(powerSettings.nWeeks).keys()].map((idx) => ({
     users: (idx + 1) * powerSettings.usersPerWeek,
     metrics: {},
   }));
 
-  for (let i = 0; i < nMetrics; i++) {
-    const metricKey = metricKeys[i];
-    const thisMetric = metrics[metricKey];
-    let thisMean = 0;
-    let thisVariance = 1;
-    if (thisMetric.type === "binomial") {
-      thisMean = thisMetric.conversionRate;
-      thisVariance =
-        thisMetric.conversionRate * (1 - thisMetric.conversionRate);
-    } else {
-      thisMean = thisMetric.mean;
-      thisVariance = thisMetric.standardDeviation ** 2;
-    }
+  Object.entries(powerSettings.metrics).forEach(([metricKey, thisMetric]) => {
     let thisMDENumeric = NaN;
     let thisSampleSizeAndRuntimeNumeric = 999;
     let lookingForSampleSizeAndRunTime = true;
-    for (let j = 0; j < nWeeks; j++) {
+    for (let j = 0; j < powerSettings.nWeeks; j++) {
       const n = powerSettings.usersPerWeek * (j + 1);
-      const thisPower = powerEst(
-        thisMetric.effectSize,
-        thisMean,
-        thisVariance,
-        n,
-        powerSettings.nVariations,
-        powerSettings.alpha,
-        true,
-        sequentialTuningParameter
-      );
+      /*initialize block-scoped variables*/
+      let thisPower = 0.0;
+      let thisMde: MDEResults = {
+        type: "success",
+        mde: -999,
+      };
+      if (powerSettings.statsEngineSettings.type === "frequentist") {
+        thisPower = powerEst(
```

Comment:
Should we call these methods `powerEstFrequentist` and `findMdeFrequentist` for consistency?

Reply:
done

Reply:
not sure why there is not an option to reply to the comment below ("If it could possibly happen, then we should have proper error handling for it, if it is impossible, we should remove it."), but I have revamped the bayesian mde code, and this does not exist anymore.